src/app/GitUI/LeftPanel/CachedFavorites.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System.Text.Json;
+
+namespace GitUI.LeftPanel
+{
+    public class CachedFavorites
+    {
+        private readonly HashSet<string> _favorites = new();
```

Comment:
Using ConcurrentBag in place of HashSet is not a straightforward replacement because ConcurrentBag does not provide the same set-like behavior, such as preventing duplicates and providing efficient lookups. Also we need Lock while reading the file